src/afl-forkserver.c

```diff
@@ -1242,10 +1242,10 @@ void afl_fsrv_start(afl_forkserver_t *fsrv, char **argv,
 
 void afl_fsrv_kill(afl_forkserver_t *fsrv) {
 
-  if (fsrv->child_pid > 0) { kill(fsrv->child_pid, fsrv->kill_signal); }
+  if (fsrv->child_pid > 0) { kill(fsrv->child_pid, fsrv->child_kill_signal); }
   if (fsrv->fsrv_pid > 0) {
 
-    kill(fsrv->fsrv_pid, fsrv->kill_signal);
+    kill(fsrv->fsrv_pid, SIGTERM);
```

Comment:
> 

I would argue that then, there should be a new environment variable to set the kill signal for the fork server. Since SIGKILL is fine (for the default setup AFL++ is used for) to kill the child, it is not fine to be used to terminate the fork server because it then has no chance to reap the child.

Reply:
There should be a `fsrv.fsrv_kill_signalI` defaulting to `SIGTERM` that can be configured via an environment variable for cases as yours mentioned previously.

Reply:
maybe like this: have another variable for the forkserver, defaulting to SIGTERM. if the KILL_SIGNAL env is set, this value is set for the child and the forkserver.

The need for the KILL_SIGNAL env this is only with kvm-like setups, and there the signal is changed to a dummy signal. so and extra env to control the forkserver signal would not be needed.